### DISCUSSION TITLE
Why plugin system is designed as using schema.json?

### DISCUSSION THREAD

**Discussion by monitor1379** [apache/apisix#5127] [Q&A] [answered]
Problem: I am confused by why plugin system is designed as using file "api/conf/schema.json" instead of querying `http://{apisix_host}:{control_port}/v1/schema` when opening `http://{apsix_dashboard_host}/plugin/market` every time so can always get the newest plugin configs from apisix?

Documentation: [FAQ/#4-after-modifying-the-plugin-schema-or-creating-a-custom-plugin-in-apache-apisix-why-cant-i-find-it-on-the-dashboard](https://apisix.apache.org/docs/dashboard/FAQ/#4-after-modifying-the-plugin-schema-or-creating-a-custom-plugin-in-apache-apisix-why-cant-i-find-it-on-the-dashboard)

Code of `initSchema()`: [apisix-dashboard :: api/internal/conf/conf.go#L278-L285](https://github.com/apache/apisix-dashboard/blob/0972f1610d1c57cbf59c8069dd2acd94c23cdbd4/api/internal/conf/conf.go#L278-L285)

![Screen Shot 2021-09-23 at 6 02 15 PM](https://user-images.githubusercontent.com/8276409/134488987-a9e98b85-5239-438d-b6f1-e182bdaedcc3.png)

**Comment by tokers** [comment, marked as answer]
The Control Plane itself shouldn't reply on an API exposed by the Data Plane, and CP doesn't have to know where the DP(s) are deployed.

Also, the Control APIs are designed to make some ops operation easy and they are helpful for troubleshooting.

**Reply by monitor1379** [reply]
Looks reasonable, now it's clear for me. thanks for answering! :D
